File: chattools/src/main/java/aytackydln/chattools/telegram/dto/models/KeyboardItem.java
package aytackydln.chattools.telegram.dto.models;

import com.fasterxml.jackson.databind.PropertyNamingStrategy;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.RequiredArgsConstructor;

@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
@Data
@RequiredArgsConstructor
@AllArgsConstructor
public class KeyboardItem{
	private String text;
	private String callbackData;

	public KeyboardItem(String text){
		this.text=text;
	}

	public String getJson(){
		return "[{ \"text\": \""+text+"\", \"callback_data\": \""+callbackData+"\" }],";
	}
}

File: chattools/src/main/kotlin/aytackydln/chattools/telegram/dto/models/KeyboardItem.kt
package aytackydln.chattools.telegram.dto.models

import com.fasterxml.jackson.databind.PropertyNamingStrategies
import com.fasterxml.jackson.databind.annotation.JsonNaming

@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy::class)
data class KeyboardItem(
    var text: String,
    var callbackData: String? = null,
)
